src/pipeline.rs

```diff
@@ -397,41 +371,69 @@ impl Device {
             shader
         };
 
+        Ok(Shader(shader))
+    }
+
+    /// Create a new shader from GLSL.
+    ///
+    /// # Valid usage
+    ///
+    /// - `source` must be a NULL-terminated C-String.
+    /// - The GLSL shader version must be `450 core` or higher.
+    /// - The `stage` parameter must be a valid stage of the passed shader source.
+    pub unsafe fn create_shader(&self, stage: ShaderStage, source: &[u8]) -> Result<Shader> {
+        let shader = self.compile_shader(stage, source)?;
+
+        let log = self.get_shader_compile_log(shader);
+        if log.is_err() {
+            println!("Shader could not be compiled successfully ({:?})", stage);
+        }
+        match log {
+            Ok(s) | Err(s) => {
+                if !s.is_empty() {
+                    println!("Shader Info Log: {}", s);
+                }
+            }
+        }
+
+        Ok(shader)
+    }
+
+    /// Return the compilation result of compiling a shader.
+    pub unsafe fn get_shader_compile_log(&self, shader: Shader) -> std::result::Result<String, String> {
         let status = {
```

Comment:
Would you mind moving the status checks into the shader/pipeline creation method?
I didn't impl it properly back then by only printing out an error message when status was false - should actually return an `Err`.
This would require two new error cases: `Compile` and `Linking`